src/client.rs

```diff
@@ -27,32 +28,22 @@ struct GraphQLResponse<T> {
   errors: Option<Vec<GraphQLErrorMessage>>,
 }
 
-impl GQLClient {
-  #[cfg(target_arch = "wasm32")]
-  fn client(&self) -> Result<reqwest::Client, GraphQLError> {
-    Ok(Client::new())
-  }
-
-  #[cfg(not(target_arch = "wasm32"))]
-  fn client(&self) -> Result<reqwest::Client, GraphQLError> {
-    Ok(
-      Client::builder()
-        .timeout(std::time::Duration::from_secs(5))
-        .build()
-        .map_err(|e| GraphQLError::with_text(format!("Can not create client: {:?}", e)))?,
-    )
-  }
-}
-
-impl GQLClient {
-  pub fn new(endpoint: impl AsRef<str>) -> Self {
+impl<'a> GQLClient<'a> {
+  pub fn new(endpoint: &'a str) -> Self {
     Self {
-      endpoint: endpoint.as_ref().to_string(),
-      header_map: HeaderMap::new(),
+      endpoint: &endpoint,
+      client: if cfg!(target_arch = "wasm32") {
+        Client::new()
+      } else {
+        Client::builder()
+          .timeout(Duration::from_secs(5))
+          .build()
+          .unwrap()
```

Comment:
Please do not use unwrap. you can use `map_err` convert to GraphQLError and throw it. the unwrap will be panic. 

like this
https://github.com/arthurkhlghatyan/gql-client-rs/blob/7b6bf56e99bc670afa17dff53fbcc84cd6fd1363/src/client.rs#L42